js/packages/openinference-instrumentation-llama-index/src/utils.ts

```diff
@@ -18,7 +17,38 @@ import {
   OpenInferenceSpanKind,
   SemanticConventions,
 } from "@arizeai/openinference-semantic-conventions";
-import { GenericFunction, SafeFunction } from "./types";
+import {
+  GenericFunction,
+  SafeFunction,
+  ObjectWithModel,
+  RetrieverQueryEngineQueryMethodType,
+  RetrieverRetrieveMethodType,
+  QueryEmbeddingMethodType,
+} from "./types";
+import { BaseEmbedding, BaseRetriever } from "llamaindex";
+
+/**
+ * Checks if
+ * @param prototype -
+ * @returns
+ */
+export function isRetriever(
+  moduleClassPrototype: unknown,
+): moduleClassPrototype is BaseRetriever {
+  return (
+    !!moduleClassPrototype && !!(moduleClassPrototype as BaseRetriever).retrieve
+  );
+}
```

Comment:
```suggestion
/**
 * Checks if the given class instance is a retriever
 */
export function isRetriever(cls: unknown): cls is BaseRetriever {
  return (
    cls != null &&
    typeof cls === "object" &&
    "retrieve" in cls &&
    typeof cls.retrieve === "function"
  );
}
```

Reply:
oh you care actually checking the class prototype. Okay let's maybe pair on this